File: src/go/fiber/internal/models/order_item.go
package models

import "time"

// OrderItem represents an item in an order
type OrderItem struct {
	ID          int       `json:"id"`
	OrderID     int       `json:"order_id"`
	ProductName string    `json:"product_name"`
	Quantity    int       `json:"quantity"`
	Price       float64   `json:"price"`
	CreatedAt   time.Time `json:"created_at"`
}

// NewOrderItem creates a new OrderItem instance
func NewOrderItem(id, orderID int, productName string, quantity int, price float64, createdAt time.Time) *OrderItem {
	return &OrderItem{
		ID:          id,
		OrderID:     orderID,
		ProductName: productName,
		Quantity:    quantity,
		Price:       price,
		CreatedAt:   createdAt,
	}
}

// String returns a string representation of the OrderItem
func (oi *OrderItem) String() string {
	return "OrderItem{ID: " + string(rune(oi.ID)) + ", Product: " + oi.ProductName + "}"
}

File: src/kotlin/ktor/src/main/kotlin/com/benchmark/models/OrderItem.kt
package com.benchmark.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OrderItem(
    val id: Int,
    @SerialName("order_id")
    val orderId: Int,
    @SerialName("product_name")
    val productName: String,
    val quantity: Int,
    val price: Double,
    @SerialName("created_at")
    val createdAt: String
)
